docs/guides/composing-messages.md

```diff
@@ -69,4 +69,147 @@ ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> r
 );
 ```
 
-You can use this way also for building a message for Incoming Webhooks and `response_url` calls.
\ No newline at end of file
+You can use this way also for building a message for Incoming Webhooks and `response_url` calls.
+
+---
+## Block Kit Kotlin DSL
+
+If you're using Kotlin, the Java Slack SDK also provides Kotlin-style builders which you can use to build your block kit structures.
+
+```kotlin
+import com.slack.api.model.block.Blocks.*
+import com.slack.api.model.kotlin_extension.request.chat.blocks
+
+val response = slack.methods(token).chatPostMessage { req -> req
+    .channel("C1234567")
+    .blocks {
+        section {
+            // "text" fields can be constructed via plainText() and markdownText()
+            markdownText("*Please select a restaurant:*")
+        }
+        divider()
+        actions {
+            // To align with the JSON structure, you could put the elements { } block around the buttons but for brevity it can be omitted
+            // The same is true for things such as the section block's "accessory" container
+            button {
+                // For instances where only plain text is acceptable, the field's name can be filled with plain text inputs
+                text("Farmhouse", emoji = true)
+                value("v1")
+            }
+            button {
+                text("Kin Khao", emoji = true)
+                value("v2")
+            }
+        }
+    }
+}
+```
+
+### Installation
+
+You can add the Block Kit Kotlin DSL via 2 artifacts:
+
+  * The slack model kotlin extension, which adds the Kotlin DSL itself as well as the standalone `withBlocks { }` builder
+  * The slack client kotlin extension, which adds the `.blocks { }` extension function to various Slack Client builders for seamless use of the DSL with the Java builders
+    * The `.blocks { }` extension function is available on the following Java builders:
+      * ChatPostEphemeralRequestBuilder
+      * ChatPostMessageRequestBuilder
+      * ChatScheduleMessageRequestBuilder
+      * ChatUpdateRequestBuilder
+
+**Adding via Gradle:**
+
+```groovy
+dependencies {
+    implementation "com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}"
+    implementation "com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}"
+}
+```
+
+**Adding via Gradle Kotlin DSL:**
+
+```kotlin
+dependencies {
+    implementation("com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}")
+    implementation("com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}")
+}
+```
+
+### Notable examples and features
+
+**Standalone withBlocks builder which comes with the model extension**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+val constructedMessage = withBlocks {
```

Comment:
> constructedMessage 

`withBlocks` returns `blocks`, not a message. This is the same for the following examples.